pkg/apis/options/legacy_options.go

```diff
@@ -481,6 +481,7 @@ type LegacyProvider struct {
 
 	KeycloakGroups           []string `flag:"keycloak-group" cfg:"keycloak_groups"`
 	AzureTenant              string   `flag:"azure-tenant" cfg:"azure_tenant"`
+	AzureGroupField          string   `flag:"azure-group-field" cfg:"azure_group_field"`
```

Comment:
Why not use the standard group claim option we already have as part of the OIDC provider? I would rather not make provider specific fields if we don't need to

Reply:
`AzureGroupField` doesn't represent the group claim from the token. It configures the group field to be used when building the groups list(`id` or `displayName`. Default is `id`) from Microsoft Graph(available only for v2.0 oidc url). Based on this value, the `allowed-group` config values should be adjusted accordingly. If using `id` as group field, `allowed-group` should contains groups IDs, if using `displayName` as group field, `allowed-group` should contains groups name"